sbin/build.sh

```diff
@@ -875,7 +876,19 @@ generateSBoM() {
   # Below add property to metadata
   # Add OS full version (Kernel is covered in the first field)
   addSBOMMetadataProperty "${javaHome}" "${classpath}" "${sbomJson}" "OS version" "${BUILD_CONFIG[OS_FULL_VERSION]^}"
-  addSBOMMetadataProperty "${javaHome}" "${classpath}" "${sbomJson}" "OS architecture" "${BUILD_CONFIG[OS_ARCHITECTURE]^}"
+  # TODO: Replace this "if" with its predecessor (commented out below) once 
+  # OS_ARCHITECTURE has been replaced by the new target architecture variable.
+  # This is because OS_ARCHITECTURE is currently the build arch, not the target arch,
+  # and that confuses things when cross-compiling an x64 mac build on arm mac.
+  #   addSBOMMetadataProperty "${javaHome}" "${classpath}" "${sbomJson}" "OS architecture" "${BUILD_CONFIG[OS_ARCHITECTURE]^}"
+  echo "DEBUG 1"
+  if [[ "${BUILD_CONFIG[TARGET_FILE_NAME]}" =~ .*_x64_.* ]]; then
```

Comment:
I feel there should be a better thing to check here, but I'm ok with it for now :-)

Reply:
I agree that if this was not a temporary fix, the name of the target should not be used in this manner. 

Will create the issue to replace OS_ARCHITECTURE now, so ideally this code will be removed long before the March release rolls around.